### PR TITLE
feat(metrics): Add opencost_build_info metric

### LINKED ISSUE #1576: Add opencost_build_info prometheus metric
**Is your feature request related to a problem? Please describe.**
Currently there is no way of determining what version of `opencost` via PromQL. This is important for two reasons:
1. Internally we rely upon querying `${SERVICE}_build_info` to determine if a new version is available for us to upgrade to
2. When having _many_ `opencost` deployments, it's helpful to understand which cluster has which version of `opencost` running

**Describe the solution you'd like**
We should export an addition metric `opencost_build_info` that has at least a `version` label that is inferred at build time of the container. 

**Describe alternatives you've considered**
We could manually release upgrades, but that's tedious and goes against our internal goal of staying relatively up to do. 
We could also write custom scripts to determine the versions per cluster, but again would prefer for this to align with our existing dashboards.

### PR DESCRIPTION
**Pull request by Pokom**
## What does this PR change?
* Adds a `opencost_build_info` metric with a `version` label

## Does this PR relate to any other PRs?
* No

## How will this PR impact users?
* Limited impact, will add one more metric

## Does this PR address any GitHub or Zendesk issues?
* Closes #1576 

## How was this PR tested?
* Locally

## Does this PR require changes to documentation?
* N/A

## Have you labeled this PR and its corresponding Issue as "next release" if it should be part of the next Opencost release? If not, why not?

I'm looking for some input from other maintainers on:
- How to best get the version information(Git SHA, ldflags set a build time etc..)
- Is this the right place into the codebase to set the build information?

### CODE REVIEW

**Comment by AjayTripathy** [comment]
https://github.com/opencost/opencost/blob/v1.99/pkg/env/costmodelenv.go#L170 Just a note that the env variable of published version seems relevant!

**Comment by Pokom** [comment]
Thanks for the nudge @AjayTripathy! After looking through, I think the two version variables *should* be the same, but I am hesitant to rely upon one that can be overridden by users for the `opencost_build_info` metric. I feel it would be pretty easy to set that environment variable at runtime(IE, helm/jsonnet/manually) and then forget about it. 

Out of curiosity, what is the use case for being able to override `APP_VERSION`?

**Comment by Pokom** [comment]
@AjayTripathy I think this PR is ready for review at this point. It should be good enough for a first pass to start exporting the version which will be enough for me to setup rolling out `opencost` in an automated fashion. Let me know if there's anything you'd like to see tested prior to a review.

**Comment by AjayTripathy** [comment]
SGTM will take a look @Pokom ! Just curious how are you planning to automate the rollouts?

File: pkg/metrics/telemetry.go [hunk 1]
```diff
@@ -42,7 +51,7 @@ func InitKubecostTelemetry(config *MetricsConfig) {
 			Help: "kubecost_http_response_size_bytes Response size in bytes",
 		}, []string{"handler", "method", "code"})
 
-		prometheus.MustRegister(requestsCount, responseTime, responseSize)
+		prometheus.MustRegister(requestsCount, responseTime, responseSize, buildInfo)
```
Comment (AjayTripathy) [inline pkg/metrics/telemetry.go:54]: Did you mean to do this on the requestsCounts metric? I don't think that's necessary.

**Review by AjayTripathy — CHANGES_REQUESTED** [review]

**Comment by AjayTripathy** [comment]
@Pokom quick question? I think the metric got accidentaly added as a label on another request?

File: pkg/metrics/telemetry.go (hunk 1 above)
Comment (Pokom) [reply pkg/metrics/telemetry.go:54]: @AjayTripathy I didn't think this impacted the other metrics. I was piggy backing off of the existing `prometheus.MustRegister` calls . If needed, I could move out buildInfo to it's own separate `prometheus.MustRegister` call

**Comment by Pokom** [comment]
> @AjayTripathy Just curious how are you planning to automate the rollouts?

We have internal tooling setup to periodically check if there's a new container image in the registry by querying prometheus for the specific `*_build_info` metric and then does some GitOps magic under the hood to stage a rollout. 

>  I think the metric got accidentaly added as a label on another request?

Provided my response inline. I can double check the exported metrics, but I thought that `prometheus.MustRegister` accepted a variable amount of collectors and registered them independently. I don't think it will impact the other metrics.

File: pkg/metrics/telemetry.go (hunk 1 above)
Comment (AjayTripathy) [reply pkg/metrics/telemetry.go:54]: Oh NVM I think it's fine.

**Review by AjayTripathy — APPROVED** [review]

**Pull request merged**